File: samples-java/src/main/java/io/github/ilaborie/loops/samples/stream/Filter.java
package io.github.ilaborie.loops.samples.stream;

import io.github.ilaborie.loops.samples.Element;

import java.util.List;
import java.util.stream.Collectors;

import static io.github.ilaborie.loops.samples.Elements.getSomeElements;

public class Filter {

    public static void main(String[] args) {
        List<Element> elements = getSomeElements();
        List<Element> result = filter(elements);
        result.forEach(System.out::println);
    }

    public static List<Element> filter(List<Element> elements) {
        return elements.stream()
                .filter(elt -> elt.getName().startsWith("a"))
                .collect(Collectors.toList());
    }

}

File: samples-kotlin/src/main/kotlin/io/github/ilaborie/loops/samples/stream/filter.kt
package io.github.ilaborie.loops.samples.stream

import io.github.ilaborie.loops.samples.ElementKt
import io.github.ilaborie.loops.samples.getSomeElementsKt


fun main() {
    val elements = getSomeElementsKt()
    val result = filter(elements)
    result.forEach { println(it) }
}

fun filter(elements: List<ElementKt>): List<ElementKt> =
    elements.filter { it.name.startsWith("a") }